onnx/test/cpp/function_context_test.cc

```diff
@@ -149,7 +149,265 @@ void RegisterCustomFunctionSchema() {
+  schema.SetName(op_type)
+      .SetDomain(ONNX_DOMAIN)
+      .SinceVersion(13)
+      .SetDoc("A variation of LogSoftMax that takes axes as input instead of attribute.")
+      .Attr("axis", "", AttributeProto::INT, static_cast<int64_t>(-1))
+      .Input(0, "input", "Input tensor", "T", OpSchema::Single)
+      .Output(0, "output", "Output tensor", "T", OpSchema::Single)
+      .TypeConstraint("T", {"tensor(float)", "tensor(double)"}, "Type of the input and output values")
+      .SetContextDependentFunctionBodyBuilder(
+          [](const FunctionBodyBuildContext& ctx, const OpSchema& schema, FunctionProto& functionProto) -> bool {
+            const int64_t axis = ctx.getAttribute("axis") != nullptr ? ctx.getAttribute("axis")->i() : -1;
+            FunctionBuilder builder(functionProto);
+            builder.Const1D("axes", axis)
+                .Add("X_ReduceMax = ReduceMax <keepdims = 1> (input)", "axes", std::vector<int64_t>({axis}))
+                .Add(R"(
+                    X_Sub = Sub (input, X_ReduceMax)
+                    X_Exp = Exp (X_Sub)
+                    X_ReduceSum = ReduceSum <keepdims = 1> (X_Exp, axes)
+                    X_Log = Log (X_ReduceSum)
+                    output = Sub (X_Sub, X_Log)
+                )");
+            schema.BuildFunction(functionProto);
+            return true;
+          },
+          13);
+  std::cout << "StepB\n"; 
+  if (!with_missing_version) {
+    schema.SetContextDependentFunctionBodyBuilder(
+        [](const FunctionBodyBuildContext& ctx, const OpSchema& schema, FunctionProto& functionProto) -> bool {
+          const int64_t axis = ctx.getAttribute("axis") != nullptr ? ctx.getAttribute("axis")->i() : -1;
+          FunctionBuilder builder(functionProto);
+          builder.Const1D("axes", axis).Add("X_ReduceMax = ReduceMax <keepdims = 1> (input, axes)").Add(R"(
+                    X_Sub = Sub (input, X_ReduceMax)
+                    X_Exp = Exp (X_Sub)
+                    X_ReduceSum = ReduceSum <keepdims = 1> (X_Exp, axes)
+                    X_Log = Log (X_ReduceSum)
+                    output = Sub (X_Sub, X_Log)
+                )");
+          schema.BuildFunction(functionProto);
+          return true;
+        },
+        18);
+  }
+  std::cout << "StepD\n"; 
+  ONNX_NAMESPACE::OpSchemaRegistry::OpSchemaRegisterOnce unused(schema);
+  (void)unused;
+}
+
+void BuildLogSoftmaxFunction(
+    const OpSchema& schema,
+    const std::string& op_type,
+    FunctionProto& fnProto,
+    int requested_opset_version) {
+  NodeProto nodeProto;
+  nodeProto.set_op_type(op_type);
+  nodeProto.add_input("input");
+  nodeProto.add_output("output");
+
+  TypeProto floatTypeProto;
+  floatTypeProto.mutable_tensor_type()->set_elem_type(TensorProto_DataType::TensorProto_DataType_FLOAT);
+
+  FunctionBodyBuildContextImpl ctx(nodeProto, {floatTypeProto});
+  schema.BuildContextDependentFunctionWithOpsetVersion(ctx, fnProto, requested_opset_version, "", true);
```

Comment:
As discussed elsewhere, better if the build function returns true/false/status to indicate success or failure.

Reply:
avoided exception.